wsgi.py

```diff
@@ -32,24 +28,72 @@ def wake_up():
     ai_service_id = input_data.get('ai_service', 'generic_ai')
     raw_data = input_data['data']
 
-    s3_destination = f'{AWS_BUCKET}/{data_id}/{ai_service_id}'
-
-    application.logger.info(
-        'Saving data to location:  s3://%s', s3_destination
-    )
-
-    filesystem = s3.connect(AWS_KEY, AWS_SECRET)
-    s3.save_data(filesystem, s3_destination, raw_data)
-
-    message = {
-        'message': f'AI-Ops pipeline successfull for {data_id}',
-        'url': s3_destination
+    try:
+        temp_file_name = tempfile.NamedTemporaryFile(delete=False).name
+        with tarfile.open(temp_file_name, "w:gz") as tar:
+            data = io.BytesIO(json.dumps(raw_data).encode())
+            info = tarfile.TarInfo(name=f'{ai_service_id}_{data_id}.json')
+            info.size = len(data.getvalue())
+            temp_file_name = tar.name
+            tar.addfile(info, data)
+
+    except (IOError, TypeError, tarfile.TarError) as e:
+        error_msg = 'Error during TAR.GZ creation: ' + str(e)
+        ROOT_LOGGER.exception("Exception: %s", error_msg)
+        return jsonify(
+            status='Error',
+            type=str(e.__class__.__name__),
+            message=error_msg
+        ), 500
+
+    files = {
+        'upload': (
+            temp_file_name, open(temp_file_name, 'rb'),
+            'application/vnd.redhat.aiopspublisher.aiservice+tgz'
+        )
     }
 
-    application.logger.info('Publishing message on topic %s', TOPIC)
-    producer.publish_message(SERVER, TOPIC, message)
-
-    return jsonify(status='OK', message='Data published')
+    headers = {'x-rh-insights-request-id': data_id}
+
+    # send a POST request to upload service with files and headers info
+    try:
+        response = requests.post(
+            f'http://{UPLOAD_SERVICE_ENDPOINT}',
+            files=files,
+            headers=headers
+        )
+        response.raise_for_status()
+
+        try:
```

Comment:
This should be moved outside the first `try` block. No need for nesting. Current approach is really bad for readability.

I suggest to have a separate:
1. `try/except` for the HTTP Post
2. `try/except` for the file removal

both blocks are on the same level

Reply:
Addressed in the last commit